### PR TITLE
ref(TS) Convert some files to TS

### PR DESCRIPTION
**Pull request by robertpin**

### CODE REVIEW

File: react/features/chat/middleware.ts [hunk 1]
```diff
@@ -168,7 +166,7 @@ MiddlewareRegistry.register(store => next => action => {
 
     case ADD_REACTION_MESSAGE: {
         _handleReceivedMessage(store, {
-            id: localParticipant.id,
+            id: localParticipant?.id ?? '',
```
Comment (gabiborlea) [inline react/features/chat/middleware.ts:169]: maybe check for the id before calling handle function

File: react/features/chat/middleware.ts [hunk 2]
```diff
@@ -471,22 +472,23 @@ function _handleReceivedMessage({ dispatch, getState },
  * @param {boolean} isLobbyPrivateMessage - Is a lobby message.
  * @returns {void}
  */
-function _persistSentPrivateMessage({ dispatch, getState }, recipientID, message, isLobbyPrivateMessage = false) {
+function _persistSentPrivateMessage({ dispatch, getState }: IStore, recipientID: string,
+        message: string, isLobbyPrivateMessage = false) {
     const state = getState();
     const localParticipant = getLocalParticipant(state);
-    const displayName = getParticipantDisplayName(state, localParticipant.id);
+    const displayName = getParticipantDisplayName(state, localParticipant?.id ?? '');
```
Comment (gabiborlea) [inline react/features/chat/middleware.ts:479]: maybe check for id before calling function

File: react/features/jaas/functions.ts [hunk 3]
```diff
@@ -70,7 +69,7 @@ export async function sendGetDetailsRequest({ appId, baseUrl }: {
         }
 
         throw new Error('Request not successful');
-    } catch (err) {
+    } catch (err: any) {
```
Comment (gabiborlea) [inline react/features/jaas/functions.ts:72]: Isn't there a type for fetch error?

File: react/features/jaas/functions.ts [hunk 4]
```diff
@@ -111,7 +110,7 @@ export async function sendGetJWTRequest({ appId, baseUrl }: {
...
```
Comment (gabiborlea) [inline react/features/jaas/functions.ts:113]: Fetch error type?

File: react/features/jaas/functions.ts (hunk 3 above)
Comment (robertpin) [reply react/features/jaas/functions.ts:72]: TS requires that errors in catch clauses must be either `any` or `unknown`

**Review by gabiborlea — DISMISSED** [review]

File: react/features/feedback/actions.ts [hunk 5]
```diff
@@ -12,10 +12,12 @@ import {
     SUBMIT_FEEDBACK_ERROR,
     SUBMIT_FEEDBACK_SUCCESS
 } from './actionTypes';
+// eslint-disable-next-line lines-around-comment
+// @ts-ignore
 import { FeedbackDialog } from './components';
 import { sendFeedbackToJaaSRequest } from './functions';
 
-declare var config: Object;
+declare const config: IConfig;
```
Comment (saghul) [inline react/features/feedback/actions.ts:20]: Maybe this should be defined as a global.

**Review by saghul — APPROVED** [review]
LGTM with a minor comment

**Review by gabiborlea — APPROVED** [review]

**Pull request merged**